sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -382,18 +382,42 @@ private case class PostgresDialect()
       case Types.ARRAY =>
         val tableName = rsmd.getTableName(columnIdx)
         val columnName = rsmd.getColumnName(columnIdx)
-        val query =
-          s"""
-             |SELECT pg_attribute.attndims
-             |FROM pg_attribute
-             |  JOIN pg_class ON pg_attribute.attrelid = pg_class.oid
-             |  JOIN pg_namespace ON pg_class.relnamespace = pg_namespace.oid
-             |WHERE pg_class.relname = '$tableName' and pg_attribute.attname = '$columnName'
-             |""".stripMargin
+
+        /*
+         Spark does not support different dimensionality per row, therefore we retrieve the
```

Comment:
If the pgsql table was not created and inserted by Spark, it's possible that values in different rows have different dimensionality, right? In that case, is it really ok to get dimensionality from the first row?

Reply:
I would say it is ok as the read will fail anyways. Previously we were reading from metadata, and whatever dimension we get there, if in table we have a row with different dimension read will fail.